core/lib/utils/image_display/image_display.dart

```diff
@@ -51,4 +54,50 @@ class ImageGallery {
       ),
     );
   }
+
+  Stack buildPickerImage({
+    required Uint8List image,
+    required BuildContext context,
+    required double width,
+    required VoidCallback onRemoveTap,
+  }) {
+    return Stack(
+      alignment: Alignment.topRight,
+      children: [
+        GestureDetector(
+          onTap: () {
+            Navigator.of(context).push(
+              PageRouteBuilder(
+                opaque: false,
+                pageBuilder: (_, __, ___) => ImagePreviewMemory(image: image),
+              ),
+            );
+          },
+          child: ClipRRect(
+            borderRadius: BorderRadius.circular(8),
+            child: AspectRatio(
+              aspectRatio: 1,
+              child: Image.memory(
+                image,
+                fit: BoxFit.cover,
+              ),
+            ),
+          ),
+        ),
+        Padding(
```

Comment:
Consider replacing this with `ImageButton` for better functionality. With `ImageButton`, you'll get the correct mouse pointer automatically and can easily add a toast message. For consistent sizing, set `iconSize: 24` for optimal results instead of dynamically sizing it.